src/screens/Nearby/NearbyResults.tsx

```diff
@@ -1,93 +1,69 @@
-import {flatMap} from '../../utils/array';
-import {
-  formatToClockOrRelativeMinutes,
-  isInThePast,
-  isSeveralDays,
-  missingRealtimePrefix,
-} from '../../utils/date';
-import insets from '../../utils/insets';
-import {getLineNameFromEstimatedCall} from '../../utils/transportation-names';
-import {DeparturesWithStopLocal, QuayWithDeparturesAndLimits} from './utils';
+import haversineDistance from 'haversine-distance';
+import sortBy from 'lodash.sortby';
+import React, {useEffect, useState} from 'react';
+import {ActivityIndicator, LayoutAnimation, View} from 'react-native';
+import {QuayGroup, StopPlaceGroup} from '../../api/departures/types';
+import {ActionItem} from '../../components/sections';
 import ThemeText from '../../components/text';
-import ThemeIcon from '../../components/theme-icon';
-import {NearbyTexts, useTranslation, dictionary} from '../../translations';
+import {Location} from '../../favorites/types';
+import MessageBox from '../../message-box';
+import {StyleSheet, useTheme} from '../../theme';
+import {NearbyTexts, useTranslation} from '../../translations';
+import QuaySection from './section-items/quay-section';
+import {hasNoGroupsWithDepartures, hasNoQuaysWithDepartures} from './utils';
 
 type NearbyResultsProps = {
-  departures: DeparturesWithStopLocal[] | null;
-  onShowMoreOnQuay?(quayId: string): void;
+  departures: StopPlaceGroup[] | null;
+  lastUpdated?: Date;
+  currentLocation?: Location;
   isFetchingMore?: boolean;
   error?: string;
   isInitialScreen: boolean;
+
+  showOnlyFavorites: boolean;
 };
 
-const NearbyResults: React.FC<NearbyResultsProps> = ({
+export default function NearbyResults({
   departures,
-  onShowMoreOnQuay,
+  lastUpdated,
   isFetchingMore = false,
   error,
   isInitialScreen,
-}) => {
+  currentLocation,
+
+  showOnlyFavorites,
+}: NearbyResultsProps) {
   const styles = useResultsStyle();
-  const navigation = useNavigation<NearbyScreenNavigationProp>();
   const {t} = useTranslation();
-  const onPress = (departure: EstimatedCall) => {
-    const {publicCode, name} = getLineNameFromEstimatedCall(departure);
-    const title = publicCode
-      ? `${publicCode} ${name}`
-      : name
-      ? name
-      : t(dictionary.travel.line.defaultName);
-    navigation.navigate('DepartureDetailsModal', {
-      title,
-      serviceJourneyId: departure.serviceJourney.id,
-      date: departure.date,
-      fromQuayId: departure.quay?.id,
-    });
-  };
 
   if (isInitialScreen) {
     return (
       <View style={styles.container}>
         <MessageBox type="info">
-          <ThemeText>{t(NearbyTexts.results.messages.initial)}</ThemeText>
+          <ThemeText>
+            {showOnlyFavorites
+              ? t(NearbyTexts.results.messages.initial)
+              : t(NearbyTexts.results.messages.initial)}
+          </ThemeText>
         </MessageBox>
       </View>
     );
   }
 
-  if (hasNoQuays(departures)) {
+  if (hasNoQuaysWithDepartures(departures)) {
```

Comment:
Wonder if loading state should be considered here?

Reply:
Sure!